### PR TITLE
uyuni/master: Build the appendline correctly for RHEL-family <= 9 

### PR DESCRIPTION
**Pull request by agraul**
This PR is a backport of https://github.com/cobbler/cobbler/pull/3198 and https://github.com/cobbler/cobbler/pull/3335.

Bug report: [bsc#1216437](https://bugzilla.suse.com/show_bug.cgi?id=1216437)

### CODE REVIEW

File: tests/actions/buildiso/append_line_test.py [hunk 1]
```diff
@@ -32,6 +32,27 @@ def test_generate_system(
...
+    blendered_data = utils.blender(cobbler_api, False, test_system)
+    test_builder = AppendLineBuilder(test_distro.name, blendered_data)
+
+    # Act
+    result = test_builder.generate_system(test_distro, test_system, False)
+
+    # Assert
+    # Very basic test yes but this is the expected result atm
+    # TODO: Make tests more sophisticated
+    assert result == f"  APPEND initrd=/{test_distro.name}.img inst.ks=default.ks"
```
Comment (m-czernek) [inline tests/actions/buildiso/append_line_test.py:53]: Upstream (in https://github.com/cobbler/cobbler/pull/3335/files), this file contains `APPEND initrd={...}`. We seem to add `/` after `initrd=`

Is that expected?

File: tests/actions/buildiso/append_line_test.py [hunk 2]
```diff
@@ -77,3 +98,38 @@ def test_generate_profile_install(request, cobbler_api, create_distro, create_pr
...
+    blendered_data = utils.blender(cobbler_api, False, test_profile)
+    test_builder = AppendLineBuilder(test_distro.name, blendered_data)
+
+    # Act
+    result = test_builder.generate_profile("redhat", "rhel7")
+
+    # Assert
+    # Very basic test yes but this is the expected result atm
+    # TODO: Make tests more sophisticated
+    assert result == f"  APPEND initrd=/{test_distro.name}.img inst.ks=default.ks"
```
Comment (m-czernek) [inline tests/actions/buildiso/append_line_test.py:117]: Same as above; additionally, we modify upstream's `append` to ` APPEND` in these tests

File: cobbler/tftpgen.py [hunk 3]
```diff
@@ -903,9 +903,10 @@ def build_kernel_options(self, system, profile, distro, image, arch: enums.Archs
                                        % (httpserveraddress, profile.name)
 
             if distro.breed is None or distro.breed == "redhat":
-
-                append_line += " kssendmac"
-                append_line = "%s inst.ks=%s" % (append_line, autoinstall_path)
+                if distro.os_version in ["rhel4", "rhel5", "rhel6", "fedora16"]:
+                    append_line += " kssendmac ks=%s" % autoinstall_path
+                else:
+                    append_line += " inst.ks.sendmac inst.ks=%s" % autoinstall_path
```
Comment (m-czernek) [inline cobbler/tftpgen.py:909]: I expect that this is equivalent to `inst.kssendmac` from https://github.com/cobbler/cobbler/pull/3198/files ?

File: tests/actions/buildiso/append_line_test.py (hunk 1 above)
Comment (agraul) [reply tests/actions/buildiso/append_line_test.py:53]: Yes, that's due to the timing of the original PR and this backport. In the meantime, this was changed in both the upstream and in our code base. This is the upstream commit that brought the change: https://github.com/cobbler/cobbler/commit/67e8a2cafc

File: cobbler/tftpgen.py (hunk 3 above)
Comment (agraul) [reply cobbler/tftpgen.py:909]: That was changed in the second PR (https://github.com/cobbler/cobbler/pull/3335/files#diff-1637482678b3b5bc9ef6ec9b6b25f9743b9d25783ca5ee7e78f9d0964809eb88) listed in the description. The first implementation was not working for older distros.

**Review by m-czernek — APPROVED** [review]
In that case, LGTM! I can't comment on functionality, but the backport seems OK to me.

**Pull request merged**